### PR TITLE
docs: update cookbooks #138

### PR DESCRIPTION
**Pull request by alaeddine-13**
updates cookbooks
closes https://github.com/jina-ai/internal-tasks/issues/138

### CODE REVIEW

File: .github/2.0/cookbooks/Document.md [hunk 1]
```diff
@@ -262,10 +263,10 @@ d4 = Document(blob=tf.SparseTensor(indices, values, dense_shape))
 | --- | --- |
 | `doc.tags` | A structured data value, consisting of fields which map to dynamically typed values |
 | `doc.id` | A hexdigest that represents a unique Document ID |
+| `doc.parent_id` | A hexdigest that ehe document's parent id |
```
Comment (maximilianwerk) [inline .github/2.0/cookbooks/Document.md:266]: typo "ehe"

File: .github/2.0/cookbooks/Document.md [hunk 2]
```diff
@@ -467,7 +469,7 @@ You can add **chunks** (sub-Document) and **matches** (neighbour-Document) to a
   d.matches.append(Document())
   ```
 
-Note that both `doc.chunks` and `doc.matches` return `DocumentArray`, which we will introduce later.
+Note that both `doc.chunks` and `doc.matches` return `ChunkArray` and `MatchArray`, which are sub-classes of `DocumentArray`. We will introduce `DocumentArray later.
```
Comment (maximilianwerk) [inline .github/2.0/cookbooks/Document.md:472]: We could add a link to the respective section here. Is there an opinion about linking in the Cookbooks already?

File: .github/2.0/cookbooks/Document.md [hunk 3]
```diff
@@ -636,6 +638,100 @@ for evaluation_key, evaluation_score in d.evaluations.items():
  recall => recall at 10: 0.5
 ```
 
+### `GraphDocument`
+`GraphDocument` is a subclass of `Document`. It's a special type of `Document` that adds functionality to let you work with a `Document` as a `directed graph`.
+Chunks of the document represent the nodes of the graph. `GraphDocument` adds graph-specific attributes (`nodes`, `adjacency` list, `edge_features`,...) and operations (`add_node`, `remove_node`, `add_edge`, `remove_edge`,...)
+
+
+#### `GraphDocument` Additional Attributes
```
Comment (maximilianwerk) [inline .github/2.0/cookbooks/Document.md:646]: Not sure, but don't we use sentence case for headers? Whatever it is, it is not consistent between this line and the next header "####`GraphDocument` methods".

**Review by maximilianwerk — COMMENTED** [review]
Looks great. :) Some minor comments

File: .github/2.0/cookbooks/Document.md (hunk 1 above)
Comment (alaeddine-13) [reply .github/2.0/cookbooks/Document.md]: fixed

File: .github/2.0/cookbooks/Document.md (hunk 2 above)
Comment (alaeddine-13) [reply .github/2.0/cookbooks/Document.md]: done

File: .github/2.0/cookbooks/Document.md (hunk 3 above)
Comment (alaeddine-13) [reply .github/2.0/cookbooks/Document.md]: I applied sentence case for the rest of the headers and to this one too

File: .github/2.0/cookbooks/Executor.md [hunk 4]
```diff
@@ -573,7 +579,7 @@ with Flow().\
 
 ---
 
-## Migration in Practice
+## Migrating from 1.x to 2.0 in Practice
```
Comment (maximilianwerk) [inline .github/2.0/cookbooks/Executor.md:582]: Should this section be removed completely? @hanxiao

File: .github/2.0/cookbooks/Executor.md [hunk 5]
```diff
@@ -369,7 +367,7 @@ requests:
 
 - `jtype` is a string. Defines the class name, interchangeable with bang mark `!`;
 - `with` is a map. Defines kwargs of the class `__init__` method
-- `metas` is a map. Defines the meta information of that class. Compared to `1.x` it is reduced to the following fields:
+- `metas` is a map. Defines the meta information of that class. It contains the following fields:
```
Comment (cristianmtr) [inline .github/2.0/cookbooks/Executor.md:370]:
```suggestion
- `metas` is a dictionary. It defines the meta information of that class. It contains the following fields:
```

File: .github/2.0/cookbooks/Math.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
+# Jina Math
+
+Jina offers several Math helpers used by it's core logic and that users can use for their own purposes (for example, in their executors).
```
Comment (cristianmtr) [inline .github/2.0/cookbooks/Math.md:3]:
```suggestion
Jina offers several Math helpers used by its core logic that users can use for their own purposes (for example, in their Executors).
```

File: .github/2.0/cookbooks/Executor.md (hunk 4 above)
Comment (hanxiao) [reply .github/2.0/cookbooks/Executor.md:582]: yes, i'm taking this branch now

**Review by hanxiao — CHANGES_REQUESTED** [review]
i'm taking this branch now for final touch

**Pull request merged**